internal/gitstream/assign.go

```diff
@@ -10,25 +10,27 @@ import (
 	"github.com/google/go-github/v47/github"
 	"github.com/qbarrand/gitstream/internal"
 	"github.com/qbarrand/gitstream/internal/config"
+	fs "github.com/qbarrand/gitstream/internal/filesystem"
 	gh "github.com/qbarrand/gitstream/internal/github"
 	"github.com/qbarrand/gitstream/internal/gitutils"
 	"github.com/qbarrand/gitstream/internal/markup"
 	"gopkg.in/yaml.v3"
 )
 
-const ownersFile = "OWNERS"
+const ownersFile = "../../OWNERS"
 
 type Assign struct {
-	GC             *github.Client
-	DryRun         bool
-	Finder         markup.Finder
-	GitHelper      gitutils.Helper
-	Logger         logr.Logger
-	IssueHelper    gh.IssueHelper
-	UserHelper     gh.UserHelper
-	Repo           *git.Repository
-	RepoName       *gh.RepoName
-	UpstreamConfig config.Upstream
+	GC               *github.Client
+	DryRun           bool
+	Finder           markup.Finder
+	GitHelper        gitutils.Helper
+	Logger           logr.Logger
+	IssueHelper      gh.IssueHelper
+	UserHelper       gh.UserHelper
+	Repo             *git.Repository
+	RepoName         *gh.RepoName
+	UpstreamConfig   config.Upstream
+	FileSystemHelper fs.FileSystemHelper
```

Comment:
I think this should rather be a `OwnersHelper` interface with a `FromFile(path string) (*Owners, error)` method that would roughly do what `getOwnersContent` currently does. Right now I feel that it's a bit too generic. WDYT?

Reply:
WDYT about:
```
type Owners struct {
    Approvers []string `yaml:"approvers"`
    Reviewers []string `yaml:"reviewers"`
    Component string   `yaml:"component"`
}

type OwnersHelper interface {
    FromFile(path string) (*Owners, error)
    IsApprover(owners *Owners, userLogin string) bool
    GetRandomApprover(owners *Owners) (string, error)
}

type ownersHelper struct{}

func NewOwnersHelper() OwnersHelper {
    return &ownersHelper{}
}

func (oh *ownersHelper) FromFile(path string)                        {...}
func (oh *ownersHelper) IsApprover(owners *Owners, userLogin string) {...}
func (oh *ownersHelper) GetRandomApprover(owners *Owners)            {...}
```

Reply:
/lgtm 🙂

Reply:
Done.